--- ./lib/buildr/packaging/artifact.rb.orig	2010-04-02 11:01:20.000000000 -0400
+++ ./lib/buildr/packaging/artifact.rb	2010-04-16 16:38:01.287801317 -0400
@@ -341,6 +341,7 @@
 
   protected
 
+
     # :call-seq:
     #   download
     # 
@@ -356,7 +357,8 @@
       fail 'No remote repositories defined!' if remote.empty?
       exact_success = remote.find do |repo_url|
         begin
-          path = "#{group_path}/#{id}/#{version}/#{File.basename(name)}"
+          path =   Buildr.repositories.build_path(group_path, id, version, name)
+          # " #{group_path}/#{id}/#{version}/#{File.basename(name)}"
           URI.download repo_url + path, name
           true
         rescue URI::NotFoundError
@@ -438,6 +440,12 @@
   class Repositories
     include Singleton
 
+
+    def build_path(group_path, id, version, name)
+      return  "#{group_path}/#{id}/#{version}/#{File.basename(name)}"
+    end
+
+
     # :call-seq:
     #   local => path
     #
