--- emma-2.0.5312/build.xml.orig	2008-07-29 17:30:09.420133817 -0400
+++ emma-2.0.5312/build.xml	2008-07-29 17:30:37.507777367 -0400
@@ -146,28 +146,6 @@
   </target>
 
   <target name="-timestamp.1" depends="init" unless="build.is.dirty" >
-    <java classname="com.vladium.util.version.VersionStampTool"
-          classpathref="lib.internal.classpath"
-          output="${temp.dir}/.ready.build.stamp"
-          append="no"
-    >
-        <arg value="-start" />
-        <arg value="${app.project.start.date}" />
-        <arg value="-format" />
-        <arg value="~F~" />
-    </java>
-    <java classname="com.vladium.util.version.VersionStampTool"
-          classpathref="lib.internal.classpath"
-          output="${temp.dir}/timestamp.properties"
-          append="no"
-    >
-        <arg value="-start" />
-        <arg value="${app.project.start.date}" />
-        <arg value="-fend" />
-        <arg value="${temp.dir}/.ready.build.stamp" />
-        <arg value="-format" />
-        <arg value="app.build.id=${app.build.id.format}\napp.build.date=${app.build.date.format}" />
-    </java>
     <!-- TODO: this is ugly and breaks private builds (constant appending is bad any way): -->
     <!-- <echo message="file.release.prefix=${file.release.prefix}" file="${temp.dir}/timestamp.properties" append="yes" /> -->
   </target>
--- emma-2.0.5312/core/data/com/vladium/app/IAppVersion.java.orig	1969-12-31 19:00:00.000000000 -0500
+++ emma-2.0.5312/core/data/com/vladium/app/IAppVersion.java	2008-07-29 17:38:47.530675873 -0400
@@ -0,0 +1,41 @@
+/* Copyright (C) 2003 Vladimir Roubtsov. All rights reserved.
+ * 
+ * This program and the accompanying materials are made available under
+ * the terms of the Common Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/cpl-v10.html
+ * 
+ * $Id: IAppVersion.java,v 1.1.1.1 2004/05/09 16:57:28 vlad_r Exp $
+ */
+package com.vladium.app;
+
+// ----------------------------------------------------------------------------
+/**
+ * @author Vlad Roubtsov, (C) 2003
+ */
+public
+interface IAppVersion
+{
+    // public: ................................................................
+
+
+    // filled in by the build:
+    
+    int APP_MAJOR_VERSION = /* */ 2; //  */ 0;
+    int APP_MINOR_VERSION = /* */ 0; //  */ 0;
+
+    int APP_BUILD_ID = /* */ 9129; //  */ 0;
+    String APP_BUILD_RELEASE_TAG = /* */ " (unsupported private build)"; //  */ " (unsupported private build)";
+    String APP_BUILD_DATE = /* */ "2008/07/29 17:38:47"; //  */ "unknown";
+
+    String APP_BUG_REPORT_LINK = /* */ "http://sourceforge.net/projects/emma"; //  */ "this private build is unsupported";
+    String APP_HOME_SITE_LINK = /* */ "http://emma.sourceforge.net/"; //  */ "this private build is unsupported";
+
+    // derived properties [must be compile-time consts]:
+
+    String APP_BUILD_ID_AND_TAG = "" + APP_BUILD_ID + APP_BUILD_RELEASE_TAG;
+    
+    String APP_VERSION = "" + APP_MAJOR_VERSION + "." + APP_MINOR_VERSION;
+    String APP_VERSION_WITH_BUILD_ID_AND_TAG = APP_VERSION + "." + APP_BUILD_ID_AND_TAG;
+
+} // end of interface
+// ----------------------------------------------------------------------------





