--- src/java/org/apache/log4j/jmx/Agent.java.sav	2006-09-14 09:04:18.000000000 +0200
+++ src/java/org/apache/log4j/jmx/Agent.java	2007-04-18 15:40:59.000000000 +0200
@@ -19,7 +19,7 @@
 import javax.management.ObjectName;
 import javax.management.MBeanServer;
 import javax.management.MBeanServerFactory;
-import com.sun.jdmk.comm.HtmlAdaptorServer;
+import mx4j.tools.adaptor.http.HttpAdaptor;
 
 import org.apache.log4j.Logger;
 
@@ -36,7 +36,7 @@
   void start() {
 
     MBeanServer server = MBeanServerFactory.createMBeanServer();
-    HtmlAdaptorServer html = new HtmlAdaptorServer();
+    HttpAdaptor html = new HttpAdaptor(8082);
 
     try {
       log.info("Registering HtmlAdaptorServer instance.");
@@ -46,9 +46,14 @@
       server.registerMBean(hdm, new ObjectName("log4j:hiearchy=default"));
 
     } catch(Exception e) {
-      log.error("Problem while regitering MBeans instances.", e);
+      log.error("Problem while registering MBeans instances.", e);
+      return;
+    }
+    try {
+      html.start();
+    } catch(java.io.IOException ioe) {
+      log.error("Problem while starting HtmlAdaptorServer (HttpAdaptor).", ioe);
       return;
     }
-    html.start();
   }
 }
